R/derive_extreme_records.R

```diff
@@ -284,8 +277,7 @@ derive_extreme_records <- function(dataset = NULL,
   )
   assert_data_frame(
     dataset_add,
-    required_vars = expr_c(by_vars, extract_vars(order)),
-    optional = TRUE
+    required_vars = expr_c(by_vars, extract_vars(order))
```

Comment:
The default value in line 250 should be removed and the code in line 267-271 is no longer required. Also `by_vars` are not required for `dataset`.

Reply:
Updated, thanks!